pkg/apiserver/webhooks/datavolume-validate.go

```diff
@@ -498,7 +498,7 @@ func validateStorageSize(spec *cdiv1.DataVolumeSpec, field *k8sfield.Path) (*met
 			}
 			return &cause, false
 		}
-	} else if spec.Storage == nil || spec.Source.PVC == nil {
+	} else if spec.Storage == nil || (spec.Source != nil && spec.Source.PVC == nil) {
```

Comment:
We can only allow clone DVs to be without size: In this case, we would also be allowing DataVolumes without spec.Source and spec.SourceRef (externally populated).